### DISCUSSION TITLE
Polyfill/Ponyfill ResizeObserver

### DISCUSSION THREAD

**Discussion by RSchneider94** [guilhermerodz/input-otp#51] [Q&A] [answered]
Hey dear input-otp community,

We've been using this lib for around 3 weeks now and I can only thank you for such a cool library that helps in the accessibility A LOT! Really well done job on this lib.

I have just a question regarding the possibility to poly/ponyfill the `ResizeObserver`. We have in our application loggings and recently we had around 4 users affected, which were using an old Safari/iOS versions with the error `ReactRenderingError: Can't find variable: ResizeObserver`. Unfortunately they couldn't advance in our application and dropped off.

Now I'm wondering if it would be possible to add some ponyfill to the library in order to avoid such application crashes? Have some of you done this already? I was thinking about using [this one](https://github.com/juggle/resize-observer) as it's possible to [Switching between native and polyfilled versions](https://github.com/juggle/resize-observer?tab=readme-ov-file#switching-between-native-and-polyfilled-versions). Because I'd like to keep the original API if the browser supports it.

Thanks!

**Comment by RSchneider94** [comment, marked as answer]
I have already found a way to make this work using the second example from [Switching between native and polyfilled versions](https://github.com/juggle/resize-observer?tab=readme-ov-file#switching-between-native-and-polyfilled-versions). Closing this one 😄

**Comment by guilhermerodz** [comment]
Hi @RSchneider94 thank you so much for reporting this. I recognize a lot of libraries consume `ResizeObserver` and still don't polyfill them anywhere.
